### ISSUE TITLE
After runing model.eval() the loss is in-valid when using InPlaceABN

### ISSUE THREAD

**Issue by TuringKi** [mapillary/inplace_abn#168]
I use the same body of Seamseg(https://github.com/mapillary/seamseg) and FCOS head to train an object detection model on my private dataset. And I replace all the pytorch BN with InplaceABN.
I am struggling to the following inconsistent loss value problem:

model.train()
loss0 = model(img)
model.eval()
loss1 = model(img)

loss0 and loss1 are totally inconformity,i.e. loss0=0.001,loss1=1.5;

Did I miss something in the training progress?

**Comment by ducksoup** [comment]
@TuringKi Is this after training the model or before?

**Comment by TuringKi** [comment]
After training.

Now i have to freeze BN parameters and fine-tune the whole model for overcoming this problem.

**Comment by ducksoup** [comment]
@TuringKi does the same happen also with standard BN from Pytorch, or just InPlaceABN?

**Comment by TuringKi** [comment]
@ducksoup ,It happens in InplaceBN, ABN is fine.

**Comment by ducksoup** [comment]
@TuringKi I'm struggling a bit with pinpointing your problem, I'll need to ask a few more questions:
1) Are you training with InPlaceABN and testing with InPlaceABN, or taking a standard BN pre-trained model (or part of it) and loading its weights into an InPlaceABN model?
2) When you are testing to check the loss as in your first post, is `img` a real batch of data from your _training_ dataset?

**Comment by TuringKi** [comment]
1.I use InPlaceABN in training, and  tested InPlaceABN, ABN and pytorch-BN(read-out the ABN parameters),   they are all have inconsistent problems.
2. img is from val set.

However, this problem can be solved by replacing InPlaceABN with ABN or freezing  BN parameters(I implemented a dumyABN...), and fine tuning a few mini-batches.

**Comment by bluesky314** [comment]
Is this solved @TuringKi ?
